### ISSUE TITLE
Lint/DuplicateBranch: bad behavior with pattern matching

### ISSUE THREAD

**Issue by PikachuEXE** [rubocop/rubocop#12713]
## Expected behavior

In the following example, `in {error: error}` is needed to avoid `compile error (SyntaxError)` `illegal variable in alternative pattern (error)`
(Pattern from HTTPX wiki https://honeyryderchuck.gitlab.io/httpx/wiki/Error-Handling#error-pattern-matching)
Response is a `HTTPX::Response` or `HTTPX::ErrorResponse` (no attribute `status`)
Source code: https://gitlab.com/os85/httpx/-/blob/v1.2.2/lib/httpx/response.rb

```ruby
# frozen_string_literal: true

require "httpx"

res=HTTPX.get("https://nginx.org")
case res
in {status: 300...} 
  $stdout.puts "hi"
in {error: error}
  $stdout.puts "hi"
else
  $stdout.puts "bye"
end
```

## Actual behavior

```
For /Users/pikachuexe/projects/project/path: configuration from /Users/pikachuexe/projects/project/path/.rubocop.yml Inheriting configuration from /Users/pikachuexe/projects/project/path/.rubocop.base.yml
Inheriting configuration from /Users/pikachuexe/projects/project/path/.rubocop.performance.yml
Inheriting configuration from /Users/pikachuexe/projects/project/path/.rubocop.base.yml
Inheriting configuration from /Users/pikachuexe/projects/project/path/.rubocop.lint.yml
Inheriting configuration from /Users/pikachuexe/projects/project/path/.rubocop.base.yml
Inheriting configuration from /Users/pikachuexe/projects/project/path/.rubocop.security.yml
Inheriting configuration from /Users/pikachuexe/projects/project/path/.rubocop.base.yml
Inheriting configuration from /Users/pikachuexe/projects/project/path/.rubocop.rake.yml
Inheriting configuration from /Users/pikachuexe/projects/project/path/.rubocop.base.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/pikachuexe/projects/project/path/test.rb
W

Offenses:

test.rb:9:1: W: Lint/DuplicateBranch: Duplicate branch body detected.
in {error: error} ...
^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected

```

## Steps to reproduce the problem

- Save above code as `test.rb`
- Setup config to enable `Lint/DuplicateBranch`
- Run `rubocop test.rb`

## RuboCop version

```
1.60.2 (using Parser 3.3.0.5, rubocop-ast 1.30.0, running on ruby 3.3.0) +server [arm64-darwin22]
  - rubocop-performance 1.20.2
  - rubocop-rails 2.23.1
  - rubocop-rake 0.6.0

```

**Comment by jonas054** [comment]
Even though it's `in {error: error}` that's highlighted, what the cop reports here is that you have `$stdout.puts "hi"` in two branches. The problem goes away if you change one of them to `$stdout.puts "hello"` for example.

**Comment by PikachuEXE** [comment]
What I want to achieve is to `$stdout.puts "hi"` for both `{status: 300...}` & `{error: error}`
How should that be done?

**Comment by PikachuEXE** [comment]
Researched more and found out just need prefix `_`
https://womanonrails.com/ruby-pattern-matching-second-look (point 5)

So this works
```ruby
# frozen_string_literal: true

require "httpx"

def handle_response(res)
  puts res.inspect
  case res
  in {status: 300...} | {error: _err}
    $stdout.puts "hi"
    puts "Error: #{_err}" if _err
  else
    $stdout.puts "bye"
  end
end

handle_response(HTTPX.get("https://exampleasdadadadad.org"))
handle_response(HTTPX.get("https://nginx.org"))
handle_response(HTTPX.get("https://reddit.com"))
```
